engines/stark/services/userinterface.cpp

```diff
@@ -158,6 +161,13 @@ void UserInterface::backPrevScreen() {
 	changeScreen(_prevScreenNameStack.pop());
 }
 
+void UserInterface::quitToMainMenu() {
+	changeScreen(Screen::kScreenGame);
+	StarkResourceProvider->shutdown();
+	StarkResourceProvider->initGlobal();
+	changeScreen(Screen::kScreenMainMenu);
```

Comment:
It feels like here should have a ```StarkResourceProvider->setShouldRestoreCurrentState();```, but that would corrupt the game when it starts again. Kind of want to know why.

Reply:
It would make more sense if `StarkResourceProvider->initGlobal();` was not called here (nor on engine startup when going to the main menu), but when starting a new game.

`StarkResourceProvider->setShouldRestoreCurrentState();` is part of restoring a saved game. It asks the resource provider to restore the game state part that is specific to the current location. There's no need to call it here nor when starting a new game.